packages/auto-form/stories/AutoForm.stories.tsx

```diff
@@ -0,0 +1,169 @@
+    required: true,
+    secret: false,
+    type: 'string',
+  },
+  clientCertificate: {
+    componentProperty: true,
+    deprecated: false,
+    description: 'AMQ Client X.509 PEM Certificate',
+    displayName: 'Client Certificate',
+    group: 'security',
+    javaType: 'java.lang.String',
+    kind: 'property',
+    label: 'common,security',
+    order: 7,
+    relation: [
+      {
+        action: 'ENABLE',
+        when: [
+          {
+            id: 'skipCertificateCheck',
+            value: 'false',
+          },
+        ],
+      },
+    ],
+    required: false,
+    secret: false,
+    type: 'textarea',
+  },
+  clientID: {
+    componentProperty: true,
+    deprecated: false,
+    displayName: 'Client ID',
+    group: 'security',
+    javaType: 'java.lang.String',
+    kind: 'property',
+    label: 'common,security',
+    labelHint:
+      'Required for connections to close and reopen without missing messages. Connection destination must be a topic.',
+    order: 4,
+    required: false,
+    secret: false,
+    type: 'string',
+  },
+  password: {
+    componentProperty: true,
+    deprecated: false,
+    displayName: 'Password',
+    group: 'security',
+    javaType: 'java.lang.String',
+    kind: 'property',
+    label: 'common,security',
+    labelHint: 'Password for the specified user account.',
+    order: 3,
+    required: false,
+    secret: true,
+    type: 'string',
+  },
+  skipCertificateCheck: {
+    componentProperty: true,
+    defaultValue: 'false',
+    deprecated: false,
+    displayName: 'Check Certificates',
+    enum: [
+      {
+        label: 'Disable',
+        value: 'true',
+      },
+      {
+        label: 'Enable',
+        value: 'false',
+      },
+    ],
+    group: 'security',
+    javaType: 'java.lang.String',
+    kind: 'property',
+    label: 'common,security',
+    labelHint:
+      'Ensure certificate checks are enabled for secure production environments. Disable for convenience in only development environments.',
+    order: 5,
+    required: false,
+    secret: false,
+    type: 'string',
+  },
+  username: {
+    componentProperty: true,
+    deprecated: false,
+    displayName: 'User Name',
+    group: 'security',
+    javaType: 'java.lang.String',
+    kind: 'property',
+    label: 'common,security',
+    labelHint: 'Access the broker with this user’s authorization credentials.',
+    order: 2,
+    required: false,
+    secret: false,
+    type: 'string',
+  },
+} as IFormDefinition;
+
+export const initialValue = {
+  brokerUrl: 'tcp://blah',
+  password:
+    '»ENC:c4b1c3818185a78d61f31d3b2bcba791d0e1dc6c5e691f4bf25f8513d6ee999b',
+  skipCertificateCheck: 'false',
+  username: 'blah',
+};
+
+stories
+  .addDecorator(story => <StoryHelper>{story()}</StoryHelper>)
+  .add('AutoForm', () => {
+    const onSave = v => {
+      alert('Got value: ' + JSON.stringify(v, undefined, 2));
+    };
+    return (
+      <React.Fragment>
```

Comment:
Using fragment here causes an issue in the story. I think you can just delete it altogether?

<img width="779" alt="screen shot 2018-12-10 at 9 11 12 am" src="https://user-images.githubusercontent.com/5942899/49737755-a6797000-fc5b-11e8-88fb-2f12b7a026fe.png">

Reply:
Yeah, I think once I rework the API as suggested this should come out.